<a href="https://colab.research.google.com/github/sarah0211-02/Assignment2_StatisticalTreatmentForRetailDatasets/blob/main/PYTN_Assgn_2_Batch_3_SyahraniNabillaPardede.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Statistical Treatment For Retail Datasets

##Import Library

In [1]:
#import library pandas dan numpy
import numpy as np
import pandas as pd
from google.colab import files
filenya=files.upload()


Saving nyc-rolling-sales.csv to nyc-rolling-sales.csv


In [2]:
# For Descriptive Statistics #
import math
import statistics
import scipy.stats
import matplotlib as mpl
import matplotlib.pyplot as plt
import seaborn as sns
import statsmodels.api as sm

##Ambil Data

In [34]:
df = pd.read_csv("nyc-rolling-sales.csv")

# Cek Data
df.head(10)

Unnamed: 0  BOROUGH   NEIGHBORHOOD  \
0           4        1  ALPHABET CITY   
1           5        1  ALPHABET CITY   
2           6        1  ALPHABET CITY   
3           7        1  ALPHABET CITY   
4           8        1  ALPHABET CITY   
5           9        1  ALPHABET CITY   
6          10        1  ALPHABET CITY   
7          11        1  ALPHABET CITY   
8          12        1  ALPHABET CITY   
9          13        1  ALPHABET CITY   

                       BUILDING CLASS CATEGORY TAX CLASS AT PRESENT  BLOCK  \
0  07 RENTALS - WALKUP APARTMENTS                                2A    392   
1  07 RENTALS - WALKUP APARTMENTS                                 2    399   
2  07 RENTALS - WALKUP APARTMENTS                                 2    399   
3  07 RENTALS - WALKUP APARTMENTS                                2B    402   
4  07 RENTALS - WALKUP APARTMENTS                                2A    404   
5  07 RENTALS - WALKUP APARTMENTS                                 2    405   
6  07 RENTALS - WALKUP APARTMENTS                                2B    406   
7  07 RENTALS - WALKUP APARTMENTS                                 2    407   
8  08 RENTALS - ELEVATOR APARTMENTS                               2    379   
9  08 RENTALS - ELEVATOR APARTMENTS                               2    387   

   LOT EASE-MENT BUILDING CLASS AT PRESENT                 ADDRESS  ...  \
0    6                                  C2            153 AVENUE B  ...   
1   26                                  C7   234 EAST 4TH   STREET  ...   
2   39                                  C7   197 EAST 3RD   STREET  ...   
3   21                                  C4     154 EAST 7TH STREET  ...   
4   55                                  C2  301 EAST 10TH   STREET  ...   
5   16                                  C4  516 EAST 12TH   STREET  ...   
6   32                                  C4            210 AVENUE B  ...   
7   18                                  C7  520 EAST 14TH   STREET  ...   
8   34                                  D5            141 AVENUE D  ...   
9  153                                  D9     629 EAST 5TH STREET  ...   

  RESIDENTIAL UNITS  COMMERCIAL UNITS  TOTAL UNITS  LAND SQUARE FEET  \
0                 5                 0            5              1633   
1                28                 3           31              4616   
2                16                 1           17              2212   
3                10                 0           10              2272   
4                 6                 0            6              2369   
5                20                 0           20              2581   
6                 8                 0            8              1750   
7                44                 2           46              5163   
8                15                 0           15              1534   
9                24                 0           24              4489   

   GROSS SQUARE FEET YEAR BUILT TAX CLASS AT TIME OF SALE  \
0               6440       1900                         2   
1              18690       1900                         2   
2               7803       1900                         2   
3               6794       1913                         2   
4               4615       1900                         2   
5               9730       1900                         2   
6               4226       1920                         2   
7              21007       1900                         2   
8               9198       1920                         2   
9              18523       1920                         2   

   BUILDING CLASS AT TIME OF SALE  SALE PRICE            SALE DATE  
0                              C2     6625000  2017-07-19 00:00:00  
1                              C7         -    2016-12-14 00:00:00  
2                              C7         -    2016-12-09 00:00:00  
3                              C4     3936272  2016-09-23 00:00:00  
4                              C2     8000000  2016-11-1

In [35]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 84548 entries, 0 to 84547
Data columns (total 22 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   Unnamed: 0                      84548 non-null  int64 
 1   BOROUGH                         84548 non-null  int64 
 2   NEIGHBORHOOD                    84548 non-null  object
 3   BUILDING CLASS CATEGORY         84548 non-null  object
 4   TAX CLASS AT PRESENT            84548 non-null  object
 5   BLOCK                           84548 non-null  int64 
 6   LOT                             84548 non-null  int64 
 7   EASE-MENT                       84548 non-null  object
 8   BUILDING CLASS AT PRESENT       84548 non-null  object
 9   ADDRESS                         84548 non-null  object
 10  APARTMENT NUMBER                84548 non-null  object
 11  ZIP CODE                        84548 non-null  int64 
 12  RESIDENTIAL UNITS               84548 non-null

In [5]:
df.shape

(84548, 22)

**Keterangan**

Selanjutnya adalah menghapus kolom EASE-MENT dan Unnamed: 0, dikarenakan data tersebut tidak diperlukan dan tidak ada nilai di dalam kolom tersebut, maka agar mendapatkan hasil yang akurat, maka 2 kolom tersebut dihapus dari tabel asli

In [6]:
# Menghapus Kolom EASE-MENT dan Unnamed: 0, dikarenakan data tidak terlalu diperlukan
df_no_east = df.drop(["Unnamed: 0", "EASE-MENT"], axis=1)

# Cek Data
df_no_east.head(5)

BOROUGH   NEIGHBORHOOD                      BUILDING CLASS CATEGORY  \
0        1  ALPHABET CITY  07 RENTALS - WALKUP APARTMENTS                
1        1  ALPHABET CITY  07 RENTALS - WALKUP APARTMENTS                
2        1  ALPHABET CITY  07 RENTALS - WALKUP APARTMENTS                
3        1  ALPHABET CITY  07 RENTALS - WALKUP APARTMENTS                
4        1  ALPHABET CITY  07 RENTALS - WALKUP APARTMENTS                

  TAX CLASS AT PRESENT  BLOCK  LOT BUILDING CLASS AT PRESENT  \
0                   2A    392    6                        C2   
1                    2    399   26                        C7   
2                    2    399   39                        C7   
3                   2B    402   21                        C4   
4                   2A    404   55                        C2   

                  ADDRESS APARTMENT NUMBER  ZIP CODE  RESIDENTIAL UNITS  \
0            153 AVENUE B                      10009                  5   
1   234 EAST 4TH   STREET                      10009                 28   
2   197 EAST 3RD   STREET                      10009                 16   
3     154 EAST 7TH STREET                      10009                 10   
4  301 EAST 10TH   STREET                      10009                  6   

   COMMERCIAL UNITS  TOTAL UNITS LAND SQUARE FEET GROSS SQUARE FEET  \
0                 0            5             1633              6440   
1                 3           31             4616             18690   
2                 1           17             2212              7803   
3                 0           10             2272              6794   
4                 0            6             2369              4615   

   YEAR BUILT  TAX CLASS AT TIME OF SALE BUILDING CLASS AT TIME OF SALE  \
0        1900                          2                             C2   
1        1900                          2                             C7   
2        1900                          2                             C7   
3        1913                          2                             C4   
4        1900                          2                             C2   

  SALE PRICE            SALE DATE  
0    6625000  2017-07-19 00:00:00  
1        -    2016-12-14 00:00:00  
2        -    2016-12-09 00:00:00  
3    3936272  2016-09-23 00:00:00  
4    8000000  2016-11-17 00:00:00

In [7]:
df_no_east.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 84548 entries, 0 to 84547
Data columns (total 20 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   BOROUGH                         84548 non-null  int64 
 1   NEIGHBORHOOD                    84548 non-null  object
 2   BUILDING CLASS CATEGORY         84548 non-null  object
 3   TAX CLASS AT PRESENT            84548 non-null  object
 4   BLOCK                           84548 non-null  int64 
 5   LOT                             84548 non-null  int64 
 6   BUILDING CLASS AT PRESENT       84548 non-null  object
 7   ADDRESS                         84548 non-null  object
 8   APARTMENT NUMBER                84548 non-null  object
 9   ZIP CODE                        84548 non-null  int64 
 10  RESIDENTIAL UNITS               84548 non-null  int64 
 11  COMMERCIAL UNITS                84548 non-null  int64 
 12  TOTAL UNITS                     84548 non-null

**Keterangan**

Mengecek tipe data setiap kolom, agar lebih mempermudah untuk mengelompokkan setiap kolom dengan tipe data yang sama dan dengan tujuan untuk mempermudah dalam pengerjaan

In [36]:
# Mengelompokan data dengan type data object
data_cat_obj = [
    "NEIGHBORHOOD", 
    "BUILDING CLASS CATEGORY", 
    "TAX CLASS AT PRESENT",
    "BLOCK",
    "LOT",
    "BUILDING CLASS AT PRESENT",
    "ADDRESS",
    "APARTMENT NUMBER",
    "BUILDING CLASS AT TIME OF SALE",
    "SALE DATE"
]
for data_group in data_cat_obj:
    df_no_east[data_group] = df_no_east[data_group].astype('category')

# Mengelompokan data dengan type data integer dan mengubahnya ke numeric
data_cat_int = [
    "BOROUGH",
    "SALE PRICE",
    "LAND SQUARE FEET",
    "GROSS SQUARE FEET",
    
]
for data_group in data_cat_int:
    df_no_east[data_group] = pd.to_numeric(df_no_east[data_group], errors='coerce')

# Mengubah type data SALE DATE dari Object menjadi Date
df_no_east["SALE DATE"] = pd.to_datetime(df_no_east["SALE DATE"])

**Catatan**

Untuk menghindari Error yang terjadi pada kolom SALE PRICE yang memiliki nilai (-), maka bisa mengabaikannya dengan menambahkan errors="coerce", yang dimana nantinya isi yang memiliki (-) akan diubah menjadi NaN

**More Information**

Parameters argscalar, list, tuple, 1-d array, or Series Argument to be converted.

errors{‘ignore’, ‘raise’, ‘coerce’}, default ‘raise’

* If ‘raise’, then invalid parsing will raise an exception.
* If ‘coerce’, then invalid parsing will be set as NaN.
* If ‘ignore’, then invalid parsing will return the input.

**Keterangan**

Selanjutnya adalah mengecek jumlah data yang memliki nilai null

In [37]:
# Mengecek nilai null
df_no_east.isnull().sum()

BOROUGH                               0
NEIGHBORHOOD                          0
BUILDING CLASS CATEGORY               0
TAX CLASS AT PRESENT                  0
BLOCK                                 0
LOT                                   0
BUILDING CLASS AT PRESENT             0
ADDRESS                               0
APARTMENT NUMBER                      0
ZIP CODE                              0
RESIDENTIAL UNITS                     0
COMMERCIAL UNITS                      0
TOTAL UNITS                           0
LAND SQUARE FEET                  26252
GROSS SQUARE FEET                 27612
YEAR BUILT                            0
TAX CLASS AT TIME OF SALE             0
BUILDING CLASS AT TIME OF SALE        0
SALE PRICE                        14561
SALE DATE                             0
dtype: int64

**Keterangan**

1. Dari hasil pengecekan tersebut terdapat nilai null pada koilom "LAND SQUARE FEET sebanyak = 26252, GROSS SQUARE FEET sebanyak : 27612, dan SALE PRICE sebanyak : 14561"

2. Dari jumlah nilai null sebanyak itu maka kita dapat mengganti nilai null tersebut dengan menggunakan MEAN dari setiap kolom tersebut agar data yang dihasilkan lebih akurat

In [38]:
df_no_east['LAND SQUARE FEET'].fillna(df_no_east['LAND SQUARE FEET'].mean(), inplace=True)
df_no_east['GROSS SQUARE FEET'].fillna(df_no_east['GROSS SQUARE FEET'].mean(), inplace=True)
df_no_east['SALE PRICE'].fillna(df_no_east['SALE PRICE'].mean(), inplace=True)

In [39]:
df_no_east[['LAND SQUARE FEET', 'GROSS SQUARE FEET', 'SALE PRICE']].isnull().sum()

LAND SQUARE FEET     0
GROSS SQUARE FEET    0
SALE PRICE           0
dtype: int64

**Keterangan**

Dari pengisian nilai null tersebut dengan menggunakan mean dari setiap kolom maka di dapatkan hasil berupa jumlah null dari kolom tersebut berjumlah 0

In [15]:
df_no_east.head()

BOROUGH   NEIGHBORHOOD                      BUILDING CLASS CATEGORY  \
0        1  ALPHABET CITY  07 RENTALS - WALKUP APARTMENTS                
1        1  ALPHABET CITY  07 RENTALS - WALKUP APARTMENTS                
2        1  ALPHABET CITY  07 RENTALS - WALKUP APARTMENTS                
3        1  ALPHABET CITY  07 RENTALS - WALKUP APARTMENTS                
4        1  ALPHABET CITY  07 RENTALS - WALKUP APARTMENTS                

  TAX CLASS AT PRESENT  BLOCK  LOT BUILDING CLASS AT PRESENT  \
0                   2A    392    6                        C2   
1                    2    399   26                        C7   
2                    2    399   39                        C7   
3                   2B    402   21                        C4   
4                   2A    404   55                        C2   

                  ADDRESS APARTMENT NUMBER  ZIP CODE  RESIDENTIAL UNITS  \
0            153 AVENUE B                      10009                  5   
1   234 EAST 4TH   STREET                      10009                 28   
2   197 EAST 3RD   STREET                      10009                 16   
3     154 EAST 7TH STREET                      10009                 10   
4  301 EAST 10TH   STREET                      10009                  6   

   COMMERCIAL UNITS  TOTAL UNITS LAND SQUARE FEET GROSS SQUARE FEET  \
0                 0            5             1633              6440   
1                 3           31             4616             18690   
2                 1           17             2212              7803   
3                 0           10             2272              6794   
4                 0            6             2369              4615   

   YEAR BUILT  TAX CLASS AT TIME OF SALE BUILDING CLASS AT TIME OF SALE  \
0        1900                          2                             C2   
1        1900                          2                             C7   
2        1900                          2                             C7   
3        1913                          2                             C4   
4        1900                          2                             C2   

  SALE PRICE            SALE DATE  
0    6625000  2017-07-19 00:00:00  
1        -    2016-12-14 00:00:00  
2        -    2016-12-09 00:00:00  
3    3936272  2016-09-23 00:00:00  
4    8000000  2016-11-17 00:00:00

In [16]:
df_no_east['RESIDENTIAL UNITS'].value_counts()

1      34722
0      24783
2      16049
3       4608
4       1346
       ...  
179        1
75         1
193        1
153        1
128        1
Name: RESIDENTIAL UNITS, Length: 176, dtype: int64

**Keterangan**

Selanjutnya adalah melakukan ekplorasi data yang ada dengan menggunakan .describe()

In [12]:
df_no_east[df_no_east['COMMERCIAL UNITS']==0].describe()

BOROUGH         BLOCK           LOT      ZIP CODE  \
count  79429.000000  79429.000000  79429.000000  79429.000000   
mean       3.004797   4270.584018    395.803699  10721.690503   
std        1.299437   3588.993150    671.927040   1320.121756   
min        1.000000      1.000000      1.000000      0.000000   
25%        2.000000   1330.000000     23.000000  10304.000000   
50%        3.000000   3335.000000     52.000000  11209.000000   
75%        4.000000   6354.000000   1003.000000  11357.000000   
max        5.000000  16322.000000   9106.000000  11694.000000   

       RESIDENTIAL UNITS  COMMERCIAL UNITS   TOTAL UNITS    YEAR BUILT  \
count       79429.000000           79429.0  79429.000000  79429.000000   
mean            1.717257               0.0      1.750217   1780.395787   
std            10.146362               0.0     10.142353    551.999810   
min             0.000000               0.0      0.000000      0.000000   
25%             0.000000               0.0      1.000000   1920.000000   
50%             1.000000               0.0      1.000000   1940.000000   
75%             2.000000               0.0      2.000000   1967.000000   
max           889.000000               0.0    889.000000   2017.000000   

       TAX CLASS AT TIME OF SALE  
count               79429.000000  
mean                    1.584258  
std                     0.720488  
min                     1.000000  
25%                     1.000000  
50%                     1.000000  
75%                     2.000000  
max                     4.000000

In [17]:
df_no_east[df_no_east['RESIDENTIAL UNITS']==0].describe()

BOROUGH        BLOCK           LOT      ZIP CODE  \
count  24783.000000  24783.00000  24783.000000  24783.000000   
mean       2.537869   3343.74999    285.173143  10318.242182   
std        1.335198   3086.87143    570.307565   2136.962746   
min        1.000000      1.00000      1.000000      0.000000   
25%        1.000000   1158.00000     12.000000  10023.000000   
50%        3.000000   1944.00000     38.000000  11004.000000   
75%        4.000000   5374.00000    136.000000  11354.000000   
max        5.000000  16317.00000   9056.000000  11694.000000   

       RESIDENTIAL UNITS  COMMERCIAL UNITS   TOTAL UNITS    YEAR BUILT  \
count            24783.0      24783.000000  24783.000000  24783.000000   
mean                 0.0          0.492515      0.598152   1674.012145   
std                  0.0         16.073953     16.073655    680.924383   
min                  0.0          0.000000      0.000000      0.000000   
25%                  0.0          0.000000      0.000000   1921.000000   
50%                  0.0          0.000000      0.000000   1950.000000   
75%                  0.0          0.000000      0.000000   1962.000000   
max                  0.0       2261.000000   2261.000000   2017.000000   

       TAX CLASS AT TIME OF SALE  
count               24783.000000  
mean                    2.423879  
std                     0.913323  
min                     1.000000  
25%                     2.000000  
50%                     2.000000  
75%                     2.000000  
max                     4.000000

In [18]:
df_no_east.value_counts(["BOROUGH"])

BOROUGH
4          26736
3          24047
1          18306
5           8410
2           7049
dtype: int64

In [19]:
df_no_east.groupby('BOROUGH').mean()

BLOCK         LOT      ZIP CODE  RESIDENTIAL UNITS  \
BOROUGH                                                             
1        1108.161750  749.201409   9913.852125           2.307495   
2        4201.187544  300.252518  10361.799404           3.436658   
3        4313.198653  337.821350  11073.456938           2.035639   
4        6629.870175  229.730476  11300.420669           1.701489   
5        3254.757788  203.562782  10039.674197           1.227586   

         COMMERCIAL UNITS  TOTAL UNITS   YEAR BUILT  TAX CLASS AT TIME OF SALE  
BOROUGH                                                                         
1                0.299683     2.649623  1706.773845                   2.130176  
2                0.160874     3.603774  1745.715279                   1.525606  
3                0.147960     2.220651  1767.358673                   1.661455  
4                0.206351     1.938996  1852.045631                   1.513278  
5                0.079667     1.309869  1868.961237                   1.186207

In [20]:
df_no_east.groupby('BOROUGH').apply(pd.DataFrame.mode).reset_index(drop=True)

BOROUGH             NEIGHBORHOOD  \
0        1  UPPER EAST SIDE (59-79)   
1        2                RIVERDALE   
2        3       BEDFORD STUYVESANT   
3        4           FLUSHING-NORTH   
4        5              GREAT KILLS   

                       BUILDING CLASS CATEGORY TAX CLASS AT PRESENT  BLOCK  \
0  13 CONDOS - ELEVATOR APARTMENTS                                2     16   
1  02 TWO FAMILY DWELLINGS                                        1   3944   
2  02 TWO FAMILY DWELLINGS                                        1   2135   
3  01 ONE FAMILY DWELLINGS                                        1   5066   
4  01 ONE FAMILY DWELLINGS                                        1   2450   

   LOT BUILDING CLASS AT PRESENT                 ADDRESS APARTMENT NUMBER  \
0    1                        R4  169 WEST 95TH   STREET                    
1    1                        D4     2287 JOHNSON AVENUE                    
2    1                        R4         429 KENT AVENUE                    
3    1                        A1      131-05 40TH   ROAD                    
4    1                        A5   1235 FOREST HILL ROAD                    

   ZIP CODE  RESIDENTIAL UNITS  COMMERCIAL UNITS  TOTAL UNITS  \
0     10023                  0                 0            1   
1     10462                  1                 0            1   
2     11201                  1                 0            1   
3     11354                  1                 0            1   
4     10314                  1                 0            1   

  LAND SQUARE FEET GROSS SQUARE FEET  YEAR BUILT  TAX CLASS AT TIME OF SALE  \
0              -                 -             0                          2   
1                0                 0           0                          1   
2                0                 0           0                          1   
3              -                 -          1920                          1   
4              -                 -          1970                          1   

  BUILDING CLASS AT TIME OF SALE SALE PRICE            SALE DATE  
0                             R4        -    2017-08-07 00:00:00  
1                             D4          0  2017-01-18 00:00:00  
2                             R4          0  2016-10-06 00:00:00  
3                             A1        -    2016-11-10 00:00:00  
4                             A5        -    2017-06-15 00:00:00

In [21]:
df_no_east.groupby('BOROUGH').var()

BLOCK            LOT      ZIP CODE  RESIDENTIAL UNITS  \
BOROUGH                                                                 
1        2.874003e+05  776005.293876  1.152993e+06         494.548162   
2        1.116808e+06  580605.653876  1.061562e+06         560.408622   
3        7.334254e+06  284230.040195  1.627799e+06          73.069968   
4        1.875575e+07  242618.233395  8.154062e+05         329.581619   
5        5.352743e+06  185032.066518  2.695262e+06           5.259888   

         COMMERCIAL UNITS  TOTAL UNITS     YEAR BUILT  \
BOROUGH                                                 
1               42.708328   543.256962  420445.857838   
2                0.769235   576.031263  348076.960778   
3                2.696896    75.907229  313207.618126   
4              206.971706   542.436937  181040.864946   
5                3.919446     9.045604  188223.832009   

         TAX CLASS AT TIME OF SALE  
BOROUGH                             
1                         0.301491  
2                         0.700288  
3                         0.788360  
4                         0.645719  
5                         0.409847

In [22]:
df_no_east.groupby('BOROUGH').std()

BLOCK         LOT     ZIP CODE  RESIDENTIAL UNITS  \
BOROUGH                                                            
1         536.097242  880.911627  1073.775249          22.238439   
2        1056.791427  761.974838  1030.321292          23.672951   
3        2708.182861  533.132291  1275.852225           8.548097   
4        4330.790455  492.562923   902.998467          18.154383   
5        2313.599684  430.153538  1641.725274           2.293445   

         COMMERCIAL UNITS  TOTAL UNITS  YEAR BUILT  TAX CLASS AT TIME OF SALE  
BOROUGH                                                                        
1                6.535161    23.307873  648.417965                   0.549082  
2                0.877060    24.000651  589.980475                   0.836832  
3                1.642223     8.712475  559.649549                   0.887896  
4               14.386511    23.290276  425.488972                   0.803566  
5                1.979759     3.007591  433.847706                   0.640193

In [40]:
# Melakukan Grade data
df_no_east.groupby('BOROUGH').agg({'SALE PRICE': [np.mean, np.std, np.size]})

SALE PRICE                     
                 mean           std   size
BOROUGH                                   
1        2.902477e+06  2.141811e+07  18306
2        5.901936e+05  2.783058e+06   7049
3        8.344884e+05  3.935813e+06  24047
4        9.063757e+05  2.697190e+06  26736
5        7.525600e+05  1.641480e+06   8410

In [26]:
mean_borough = 2.902477e+06
sd_borough = 2.141811e+07
n_borough = 18306
z = 1.96 # z-score untuk confidence level 95%

##Menghitung dengan memasukan rumus standard error

Rumus :

se = sd / np.sqrt(jumlah keseluruhan data)

**Keterangan:**

se = standard error
sd = standard devisiasi

In [27]:
se = sd_borough / np.sqrt(n_borough)

# Cetak Hasil
print("Hasil standard error adalah:", se)

Hasil standard error adalah: 158301.276054372


In [28]:
lcb = mean_borough - z * se
ucb = mean_borough + z * se

print("Hasil Lower limit adalah:", lcb)
print("-------------------------")
print("Hasil Upper limit adalah:", ucb)

Hasil Lower limit adalah: 2592206.4989334308
-------------------------
Hasil Upper limit adalah: 3212747.5010665692


In [41]:
# Grading Data
grade_borough_1 = df_no_east[df_no_east["BOROUGH"] == 1]
grade_borough_2 = df_no_east[df_no_east["BOROUGH"] == 3]

# Menghitung panjang nilai
# Data 1
num_grade_borough_1 = len(grade_borough_1)
# Data 2
num_grade_borough_2 = len(grade_borough_2)

# Menghitung Mean
# Data 1
mean_grade_borough_1 = grade_borough_1["SALE PRICE"].mean()
# Data 2
mean_grade_borough_2 = grade_borough_2["SALE PRICE"].mean()

# Menghitung Standard Deviasi
# Data 1
std_grade_borough_1 = grade_borough_1["SALE PRICE"].std()
# Data 2
std_grade_borough_2 = grade_borough_2["SALE PRICE"].std()

In [42]:
print("Hasil perhitungan Data 1:",num_grade_borough_1, mean_grade_borough_1, std_grade_borough_1)
print("")
print("Hasil perhitungan Data 2:",num_grade_borough_2, mean_grade_borough_2, std_grade_borough_2)

Hasil perhitungan Data 1: 18306 2902476.5473534847 21418111.934790708

Hasil perhitungan Data 2: 24047 834488.3717719465 3935813.4285697187


In [43]:
# Summary
sm.stats.ztest(grade_borough_1["SALE PRICE"].dropna(), grade_borough_2["SALE PRICE"].dropna(), alternative="two-sided")

(14.651229205435005, 1.323003784566643e-48)